### ISSUE TITLE
SyncRule purging with unrelated ImportSources

### ISSUE THREAD

**Issue by lazyfrosch** [Icinga/icingaweb2-module-director#1324]
An interesting problem regarding the (parallel) execution of 3 sync jobs.

    Cannot delete icinga_host xxx it has not been loaded from Db

This happens sporadic when running sync with the jobs daemon.

![bildschirmfoto von 2017-12-07 10-18-33](https://user-images.githubusercontent.com/121874/33723641-7a6a9f1c-db6d-11e7-8e1f-992f98560c02.png)

Problem seems that a sync with a unrelated import source tries to delete objects that were never coming from that source.

## Involved Imports

* Hosts from OTRS
  * SQL Query for Hosts in the OTRS CMDB
  * Key: fqdn (Beispiel: xxx.customer.lan)
* Devicetypes from OTRS
  * SQL Query for all Devicetypes (only the id as field)
  * Key: id (Beispiel: firewall xx 123)
* PuppetDB
  * All nodes and facts

## Involved Sync

* Hosts from Otrs Devices
  * Complex rule involving all 3 importsources and creates host objects from them
    - Locations
    - Hosts from OTRS
    - PuppetDB Facts
* Devicetype templates (Create Templates in the screenshot)
  * Simple rule, only creating host templates for all types
* Customer Templates
  * Simple rule creating customer templates

## Your Environment

(@cyberkov)

* Director version (System - About): 16729a98ecdfd1a11a3f8da54182047b756c22a5
* Icinga Web 2 version and modules (System - About): 2.4.2 (?)
* Icinga 2 version (`icinga2 --version`): 2.7.2 (?)
* Operating System and version: Ubuntu xenial

**Comment by Thomas-Gelf** [comment]
Sync jobs should not run in parallel at all

**Comment by lazyfrosch** [comment]
Yeah I misread the `fork` inside JobRunner, I'm still not sure whats going on with purging here, its odd.

**Comment by mbassett** [comment]
I am getting this periodically as well.
Cannot delete icinga_host "i-07cb8de8c46412b9e", it has not been loaded from Db

If i click check for changes on that sync job it clears the error at least for a while.  It shows up several times a day at this point, but is not breaking anything just causing the director healthcheck to warn
